tests/TestCase/Core/FunctionsTest.php

```diff
@@ -36,6 +37,34 @@
  */
 class FunctionsTest extends TestCase
 {
+    public function testPathCombine(): void
+    {
+        $this->assertSame('', pathCombine([]));
+        $this->assertSame('', pathCombine(['']));
+        $this->assertSame('', pathCombine(['', '']));
+        $this->assertSame('/', pathCombine(['/', '/']));
+
+        $this->assertSame('path/to/file', pathCombine(['path', 'to', 'file']));
+        $this->assertSame('path/to/file', pathCombine(['path/', 'to', 'file']));
+        $this->assertSame('path/to/file', pathCombine(['path', 'to/', 'file']));
+        $this->assertSame('path/to/file', pathCombine(['path/', 'to/', 'file']));
+        $this->assertSame('path/to/file', pathCombine(['path/', '/to/', 'file']));
+
+        $this->assertSame('/path/to/file', pathCombine(['/', 'path', 'to', 'file']));
+        $this->assertSame('/path/to/file', pathCombine(['/', '/path', 'to', 'file']));
+
+        $this->assertSame('/path/to/file/', pathCombine(['/path', 'to', 'file/']));
+        $this->assertSame('/path/to/file/', pathCombine(['/path', 'to', 'file', '/']));
+        $this->assertSame('/path/to/file/', pathCombine(['/path', 'to', 'file/', '/']));
+
+        $this->assertSame('/', pathCombine([''], addTrailing: true));
+        $this->assertSame('/', pathCombine(['/'], addTrailing: true));
+        $this->assertSame('/path/to/file/', pathCombine(['/path', 'to', 'file/', '/'], addTrailing: true));
+
+        $this->assertSame('/path/to\\file', pathCombine(['/', '\\path', 'to', '\\file']));
```

Comment:
This is awful, but possible on windows :grimacing:

Reply:
It's very unlikely. I was just creating test cases.

Reply:
Yes I understand, it's good to have tests for wacky use cases :)